src/main/java/me/zeroeightsix/kami/gui/hudgui/elements/world/EntityList.kt

```diff
@@ -0,0 +1,95 @@
+import me.zeroeightsix.kami.setting.GuiConfig.setting
+import me.zeroeightsix.kami.util.EntityUtils.isHostile
+import me.zeroeightsix.kami.util.EntityUtils.isNeutral
+import me.zeroeightsix.kami.util.EntityUtils.isPassive
+import me.zeroeightsix.kami.util.originalName
+import me.zeroeightsix.kami.util.threads.defaultScope
+import net.minecraft.entity.Entity
+import net.minecraft.entity.item.*
+import net.minecraft.entity.player.EntityPlayer
+import net.minecraft.entity.projectile.EntityEgg
+import net.minecraft.entity.projectile.EntitySnowball
+import net.minecraft.entity.projectile.EntityWitherSkull
+import java.util.*
+
+object EntityList : LabelHud(
+    name = "EntityList",
+    category = Category.WORLD,
+    description = "List of entities nearby"
+) {
+
+    private val item by setting("Items", true)
+    private val passive by setting("Passive Mobs", true)
+    private val neutral by setting("Neutral Mobs", true)
+    private val hostile by setting("Hostile Mobs", true)
+    private val range by setting("Range", 24.0f, 8.0f..64.0f, 0.5f)
+
+    private var cacheMap: Map<String, Int> = emptyMap()
+
+    override fun SafeClientEvent.updateText() {
+        for ((name, count) in cacheMap) {
+            displayText.add(name, primaryColor)
+            displayText.addLine("x$count", secondaryColor)
+        }
+
+        defaultScope.launch {
+            val map = TreeMap<String, Int>()
+
+            for (entity in world.loadedEntityList.toList()) {
+                if (entity == null) continue
+                if (entity == player || entity == mc.renderViewEntity) continue
+
+                if (!item && entity is EntityItem) continue
+                if (!passive && entity.isPassive) continue
+                if (!neutral && entity.isNeutral) continue
+                if (!hostile && entity.isHostile) continue
+
+                if (mc.player.getDistance(entity) > range) continue
+
+                val name = entity.entityListName
+                map[name] = map.getOrDefault(name, 0) + 1
+            }
+
+            cacheMap = map
+        }
+    }
+
+    private val Entity.entityListName
+        get() = when (this) {
+            is EntityPlayer -> {
+                "Player"
+            }
+            is EntityItem -> {
+                this.item.originalName
+            }
+            is EntityWitherSkull -> {
+                "Wither skull"
+            }
+            is EntityEnderCrystal -> {
+                "End crystal"
+            }
+            is EntityEnderPearl -> {
+                "Thrown ender pearl"
+            }
+            is EntityMinecart -> {
+                "Minecart"
+            }
+            is EntityItemFrame -> {
+                "Item frame"
+            }
+            is EntityEgg -> {
+                "Thrown egg"
+            }
+            is EntitySnowball -> {
+                "Thrown snowball"
+            }
+            else -> {
+                this.name ?: "Unknown"
```

Comment:
I'd prefer this to default to the Java class so it's still slightly useful and so we can add the name as needed

Reply:
Actually no, normally the only entity that can has `null` name is just fishing hook.

Reply:
Wtf. Okay I guess make it still print class name, so it's not entirely useless